src/app/(legal)/privacy-policy/page.tsx

```diff
@@ -1,38 +1,64 @@
+import path from 'path';
+
+import { kebabCase } from 'lodash';
+
 import Content from './content.mdx';
 
 import { ContentWithTable } from '@/components/sections/content-with-table';
+import type { ContentWithTableProps } from '@/components/sections/content-with-table/content-with-table.type';
 import { HeroPoints } from '@/components/sections/hero-points';
+import type { HeadingGroup} from '@/lib/mdx/extract-tittles';
+import { extractHeadingsFromMDX } from '@/lib/mdx/extract-tittles';
+
+
+interface HeadingGroupWithSubItems extends HeadingGroup {
+  subItems: HeadingGroup[];
+}
+
+function groupByLevel(data: HeadingGroup[]) {
+  const grouped: HeadingGroupWithSubItems[] = [];
+
+  data.forEach(item => {
+    if (item.level === 2) {
+      grouped.push({
+        ...item,
+        subItems: [],
+      });
+    } else {
+      const parentLevel2 = grouped[grouped.length - 1];
+      if (parentLevel2) {
+        parentLevel2.subItems.push(item);
+      }
+    }
+  });
+
+  return grouped;
+}
+
+function formatLinks(headings: HeadingGroupWithSubItems[]): ContentWithTableProps['links'] {
+  return headings.map(heading => ({
+    text: heading.text,
+    href: `#${kebabCase(heading.text)}`,
+    subItems: heading.subItems.map(subItem => ({
+      text: subItem.text,
+      href: `#${kebabCase(subItem.text)}`,
+    })),
+  }));
+}
```

Comment:
@RBoraurito, Won't this logic be the same on all rich text pages? Can we make it reusable?

Reply:
I think it could be reused, let me move it